### PR TITLE
[PropertyAccess] Allow to escape dots and squared brackets in property paths

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/17917

### CODE REVIEW

File: components/property_access.rst [hunk 1]
```diff
@@ -81,6 +81,20 @@ You can also use multi dimensional arrays::
     var_dump($propertyAccessor->getValue($persons, '[0][first_name]')); // 'Wouter'
     var_dump($propertyAccessor->getValue($persons, '[1][first_name]')); // 'Ryan'
 
+.. tip::
+
+    If the key of the array contains a dot ``.``, a left squared bracket ``[``
+    or a right squared bracket ``]``, it is possible to escape these characters
```
Comment (alamirault) [inline components/property_access.rst]: I checked in code, and escaping right squared bracket is "not supported" /needed
```suggestion
    If the key of the array contains a dot ``.`` or a left squared bracket, it is possible to escape these characters
```

File: components/property_access.rst [hunk 2]
```diff
@@ -81,6 +81,20 @@ You can also use multi dimensional arrays::
...
+    with a backslash. In the above example, if the array key becomes ``first.name``,
```
Comment (alamirault) [inline components/property_access.rst]:
```suggestion
    with a backslash. In the above example, if the array key ``first_name`` becomes ``first.name``,
```

**Review by alamirault — CHANGES_REQUESTED** [review]

File: components/property_access.rst (hunk 2 above)
Comment (alexandre-daubois) [reply components/property_access.rst]: Updated!

File: components/property_access.rst (hunk 1 above)
Comment (alexandre-daubois) [reply components/property_access.rst]: Did the appropriate change 👍 thanks!

**Review by alamirault — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Excellent addition as always. Thanks Alex! and thanks Antoine for the detailed review.

**Comment by chalasr** [comment]
Thank you guys, you rock.

**Comment by alanpoulain** [comment]
Thank you very much!

**Pull request merged**